src/snowflake/cli/plugins/stage/diff.py

```diff
@@ -331,3 +289,74 @@ def sync_local_diff_with_stage(
         # Could be ProgrammingError or IntegrityError from SnowflakeCursor
         log.error(err)
         raise SnowflakeSQLExecutionError()
+
+
+def _to_src_dest_pair(
+    stage_path: StagePath, bundle_map: Optional[BundleMap]
+) -> Tuple[Optional[str], str]:
+    if not bundle_map:
+        return None, str(stage_path)
+
+    dest_path = to_local_path(stage_path)
+    src = bundle_map.to_project_path(dest_path)
+    if src:
+        return str(src), str(stage_path)
+
+    return "?", str(stage_path)
+
+
+def _to_diff_line(status: str, src: Optional[str], dest: str) -> str:
+    if src is None:
+        src_prefix = ""
+    else:
+        src_prefix = f"{src} -> "
+
+    longest_status = "modified"
+    padding = " " * (len(longest_status) - len(status))
+    status_prefix = f"[red]{status}[/red]: {padding}"
+
+    return f"{status_prefix}{src_prefix}{dest}"
+
+
+def print_diff_to_console(
+    diff: DiffResult,
+    bundle_map: Optional[BundleMap] = None,
+):
+    if not diff.different and not diff.only_local and not diff.only_on_stage:
+        cc.message("Your stage is up-to-date with your local deploy root.")
+        return
+
+    blank_line_needed = False
+    if diff.only_local or diff.different:
+        cc.message("Local changes to be deployed:")
+        messages_to_output = []
+        for p in diff.different:
+            src_dest_pair = _to_src_dest_pair(p, bundle_map)
+            messages_to_output.append(
+                (
+                    src_dest_pair,
+                    _to_diff_line("modified", src_dest_pair[0], src_dest_pair[1]),
+                )
+            )
+        for p in diff.only_local:
+            src_dest_pair = _to_src_dest_pair(p, bundle_map)
+            messages_to_output.append(
+                (
+                    src_dest_pair,
+                    _to_diff_line("added", src_dest_pair[0], src_dest_pair[1]),
+                )
+            )
+
+        with cc.indented():
+            for key, message in sorted(messages_to_output, key=lambda pair: pair[0]):
+                cc.message(message)
+
+        blank_line_needed = True
+
+    if diff.only_on_stage:
+        prefix = "\n" if blank_line_needed else ""
+        cc.message(f"{prefix}Deleted paths to be removed from your stage:")
```

Comment:
this would break indentation if called inside a nested `indented()` block, how about
```suggestion
        if blank_line_needed:
            cc.message("")
        cc.message("Deleted paths to be removed from your stage:")
```

Reply:
I have a larger suggestion of updating `message()` to split lines, pad each line with indentation, then join back, but I feel like it's out of scope

Reply:
Oh wow, I can't believe I did that, thanks for catching it! As for officially supporting multiple lines, that's something we can consider as we update the rest of the messages. I do agree I've probably piled up enough content in this particular PR already though.